apps/web-giddh/src/app/permissions/components/details/permission.details.component.ts

```diff
@@ -94,8 +96,11 @@ export class PermissionDetailsComponent implements OnInit, AfterViewInit, OnDest
     * @memberof PermissionDetailsComponent
     */
     public ngAfterViewInit(): void {
-        this.pageList = this.pageList.map(item => {
-            return { label: item, value: item, additional: { isDisabled: this.checkForAlreadyExistInPageArray(String(item)) } }
+        this.pageList = [];
+        this.originalPageList?.forEach(item => {
+            if (!this.checkForAlreadyExistInPageArray(String(item))) {
```

Comment:
1) You write same code at 2 places, so use single common function 

2) why use String()